src/camp_core.F90

```diff
@@ -1456,17 +1600,16 @@ integer(kind=i_kind) function pack_size(this, comm)
     end do
     do i_sub_model = 1, size(this%sub_model)
       sub_model => this%sub_model(i_sub_model)%val
-      pack_size = pacK_size + sub_model_factory%pack_size(sub_model, l_comm)
+      pack_size = pack_size + sub_model_factory%pack_size(sub_model, l_comm)
       sub_model => null()
     end do
     pack_size = pack_size + &
                 camp_mpi_pack_size_integer(this%size_state_per_cell, l_comm) + &
-                camp_mpi_pack_size_integer(this%n_cells, l_comm) + &
                 camp_mpi_pack_size_logical(this%split_gas_aero, l_comm) + &
                 camp_mpi_pack_size_real(this%rel_tol, l_comm) + &
                 camp_mpi_pack_size_real_array(this%abs_tol, l_comm) + &
                 camp_mpi_pack_size_integer_array(this%var_type, l_comm) + &
-                camp_mpi_pack_size_real_array(this%init_state, l_comm)
+                camp_mpi_pack_size_real_array(this%init_state_cell, l_comm)
```

Comment:
Because both arrays are initialized to 1.0 at line 800:

```
          ! Set the activity coefficients to 1.0 as default
          do i_name = 1, size(unique_names)
            i_state_elem = rep%spec_state_id(unique_names(i_name)%string)
            this%init_state(i_state_elem + i_cell * this%size_state_per_cell) = &
                    real(1.0d0, kind=dp)
            this%init_state_cell(i_state_elem) = &
                    this%init_state(i_state_elem + i_cell * this%size_state_per_cell)
          end do
```

I created this "init_state_cell" to accelerate MPI communications, since I only send a cell instead of all the cells.

I believe "init_state" can be simplified quite well, but for the moment this patch is enough for me.